### PR TITLE
Added support for JWT audiences

### PR DESCRIPTION
**Pull request by umbreak**
Fixes https://github.com/BlueBrain/nexus/issues/2546

### CODE REVIEW

File: delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/identity/IdentitiesImpl.scala [hunk 1]
```diff
@@ -77,7 +81,7 @@ class IdentitiesImpl private (findActiveRealm: String => UIO[Option[Realm]], gro
       parsedToken       <- IO.fromEither(ParsedToken.fromToken(token))
       activeRealmOption <- findActiveRealm(parsedToken.issuer)
       activeRealm       <- IO.fromOption(activeRealmOption, UnknownAccessTokenIssuer)
-      _                 <- validate(parsedToken.jwtToken, realmKeyset(activeRealm))
+      _                 <- validate(activeRealm.acceptedAudiences, parsedToken.jwtToken, realmKeyset(activeRealm))
```
Comment (imsdu) [inline delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/identity/IdentitiesImpl.scala:84]: We should update`InvalidAccessToken` message to mention audiences as an error cause

**Review by imsdu — CHANGES_REQUESTED** [review]
We also need to avoid the documentation for realms

**Review by bogdanromanx — APPROVED** [review]
I would also suggest updating the message for the InvalidToken to also include a mention that the aud claim does not match expectations.

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala [hunk 2]
```diff
@@ -133,7 +135,12 @@ object RealmsRoutes {
   @nowarn("cat=unused")
   implicit final private val configuration: Configuration = Configuration.default.withStrictDecoding
 
-  final private[routes] case class RealmInput(name: Name, openIdConfig: Uri, logo: Option[Uri])
+  final private[routes] case class RealmInput(
+      name: Name,
+      openIdConfig: Uri,
+      logo: Option[Uri],
+      acceptedAudiences: Option[NonEmptySet[String]]
```
[the remark this answers is not in the document]
Comment (umbreak) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala:142]: I can restrict it to a label, but I'm not sure if it is really necessary

[the remark this answers is not in the document]
Comment (bogdanromanx) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala:142]: I don't think a label matches the audience expectation...

[the remark this answers is not in the document]
Comment (umbreak) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala:142]: The let's keep it as string

**Review by imsdu — APPROVED** [review]

**Pull request merged**